### PR TITLE
resolve unexpected behaviour created by orphaned license keys 

### LINKED ISSUE #4669: prevent unexpected behaviour which generated by orphaned license keys
# Bug Report

## User Story

As a user, I only want accurate license key errors so that the WP dashboard/admin interface is not crowded with incorrect messages.

## Details
Any user with old license keys saved in the database (a number that will grow over time as more people purchase plans to replace single add-ons and legacy bundles) gets a phantom warning related to expired/inactive license keys when in reality their plan license keys are active. 

## Current Behavior
Currently, users with old license expired license keys have no way of removing them (and thus the warnings on every admin screen) unless they know the workaround in my video below.

This creates extra work for the user, as well as a burden on the Support and Customer Success teams as they have to have access to the site to resolve the problem for the customer. 

## Expected Behavior
All licenses saved in the database should display on the licenses screen and be able to be deactivated or deleted completely.

## Bug Type
- [x] I am not sure whether this functionality ever worked as expected.

## Steps to Reproduce
The steps involve adding a live license key and then expiring it, which needs someone with EDD access to the site to facilitate. Here's a video: https://screencast-o-matic.com/watch/cYf0oAA0JT

Here's what I did before that video:
1. Add license for a single add-on to the site.
2. Log in as EDD-level admin on the givewp.com site, and manually switch that license to expired.
3. Refresh the licenses on the licenses tab of the test site.

## Related
We regularly have calls about this.

## Acceptance Criteria
- [ ] All licenses in the DB show on the back end.
- [ ] The error messages related to licenses only show if there are no active plan licenses.

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

Resolves #4669 

You can see a list of unexpected behaviors here: https://github.com/impress-org/givewp/issues/4669#issuecomment-622260708

To resolve the issue we will do following things when activating bundle plan license key:
1. Remove a single addon license key if part of the bundle plan.
2. Remove the existing bundle plan license key. Admin can only activate one bundle plan key.

Note: an automatic upgrade also added and it will only remove single addon license keys if part of an active bundle license key.

## Affects

This patch adds logic to remove single license keys and bundle plan license keys when activating the bundle plan license key. Similar logic will run on when updating the core.

## What to test

Activate multiple bundle plan license key from low to high add-on count when any of must-have add-on installed for example REcurring addon.

If you want to review automatic updates then first switch to `develop` branch. Activate the single license key (recurring, pdf receipt) and bundle plan (basic) license key. Switch to `hotfix/remove-old-single-license-key-4669`. Go to `Downloads -> Settings ->  Licenses`. Check if recurring addon still listed as licensed or not because it is not part of the basic bundle plan that why only license key for `pdf receipt` addon will be removed.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/admin/upgrades/upgrade-functions.php [hunk 1]
```diff
@@ -3593,3 +3597,30 @@ function give_v2511_upgrades() {
 		$wp_roles->remove_cap( $role, 'read_give_payment' );
 	}
 }
+
+/**
+ * Upgrade for version 2.6.3
+ *
+ * @since 2.6.3
+ */
+function give_v262_upgrades() {
```
Comment (DevinWalker) [inline includes/admin/upgrades/upgrade-functions.php:3606]: @ravinderk why is this function named `give_v262_upgrades()` when it's going into 2.6.3?

Comment (ravinderk) [reply includes/admin/upgrades/upgrade-functions.php:3606]: @DevinWalker By mistake, I name like that. I will rename it.

Comment (ravinderk) [reply includes/admin/upgrades/upgrade-functions.php]: fixed: https://github.com/impress-org/givewp/compare/84cf543e756b...9f5b722ed764

**Comment by ravinderk** [comment]
@DevinWalker This pr is ready to merge.

**Pull request merged**